src/components/TweetsList/TweetItem/index.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useCallback } from 'react';
+import { useTweets } from '../../../providers/tweets';
+
+import Card from '@material-ui/core/Card';
+import CardActionArea from '@material-ui/core/CardActionArea';
+import CardActions from '@material-ui/core/CardActions';
+import Button from '@material-ui/core/Button';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+import TextField from '@material-ui/core/TextField';
+import FormControl from '@material-ui/core/FormControl';
+
+import styles from './styles.module.scss';
+
+const TweetItem = ({ post }) => {
+  const { removeTweet, updateTweet } = useTweets();
+
+  const [isEdit, setIsEdit] = useState(false);
+  const [value, setValue] = useState(post.content);
+
+  const handleChange = (event) => {
+    event.preventDefault();
+    setValue(event.target.value);
+  };
+
+  const handleEdit = () => {
+    setIsEdit(true);
+  };
+
+  const handleSave = useCallback(
+    async () => {
+      updateTweet(value, post.id);
+      setIsEdit(false);
+    },
+    [updateTweet, post, value],
+  );
+
+  const handleCancel = useCallback(
+    () => {
+      setIsEdit(false);
+    },
+    []
+  )
+
+  const handleRemove = useCallback(
+    () => {
+      removeTweet(post.id)
+    },
+    [removeTweet, post]
+  )
+
+  return (
+    <Card className={styles.tweetItem}>
+      <CardActionArea>
+        <CardContent>
+          {
+            isEdit ? (
```

Comment:
Вот это, наверное, уместнее вынести чуть выше в переменную, так будет проще читаться:
```suggestion
            const tweetContent = editing ? (<div>Edit form</div>) : tweet.content
```

И если уж ты именуешь сущность - Tweet, то пускай она так и будет, не миксуй с `post`